addons/test/misra/misra-test.c

```diff
@@ -398,10 +398,26 @@ void misra_14_1() {
 
 }
 
-void misra_14_2() {
+void misra_14_2_init_value(int32_t *var) {
+    *var = 0;
+}
+void misra_14_2(bool b) {
   for (dostuff();a<10;a++) {} // 14.2
   for (;i++<10;) {} // 14.2
   for (;i<10;dostuff()) {} // TODO
```

Comment:
BTW: Does someone know what the `TODO` here means? Is this a false negative?
I can imagine that the MISRA addon should warn here, but I am not sure.

Reply:
I'm not completely sure that it would be MISRA violation. There are no mention of this situation in standard or our [test suite](https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/blob/master/R_14_02.c).

But I found that in [LDRA checker](https://ldra.com/aerospace-defence/products/) has the following note for MISRA R14.2 compliance:
> LDRA 271S For loop incrementation is not simple

I think that's what the author wanted to say with this `TODO`.

Reply:
The MISRA document says that the third for clause shall only change the loop counter (and nothing else that is used in the loop body).
So what I can imagine is that this is a test for the third for clause, to warn when something other than the for loop counter `i` is changed.
But since `dostuff()` is unknown it is hard to tell whether it could change anything else. It could even be some macro for changing the loop counter. So IMHO it is correct to do not warn here.

Reply:
We can say for sure that `dostuff` doesn't change anything only when we check `dostuff` scope and correlate it with `for` loop scope content. Otherwise here can be false negative.
I suggest to leave this TODO. Anyway we need same checks to detect undesirable side-effects of function call. See, for example, [unimplemented](https://trac.cppcheck.net/ticket/8667) rule 8.13, which requires it.

Reply:
> The MISRA document says that the third for clause shall only change the loop counter (and nothing else that is used in the loop body).
> So what I can imagine is that this is a test for the third for clause, to warn when something other than the for loop counter `i` is changed.
> But since `dostuff()` is unknown it is hard to tell whether it could change anything else. It could even be some macro for changing the loop counter. So IMHO it is correct to not warn here.

Reply:
Hmm, ok let's keep this line as it is. `TODO` comments without any further hint IMO are not very useful.